batch/batch/server/server.py

```diff
@@ -110,14 +113,52 @@ def _next_task(self):
         self._task_idx += 1
         if self._task_idx < len(self._tasks):
             self._current_task = self._tasks[self._task_idx]
+        else:
+            self._delete_pvc()
 
     def _has_next_task(self):
         return self._task_idx < len(self._tasks)
 
+    def pvc(self):
```

Comment:
I don't like implicit resource creation, because it's harder to reason about (since resources can effectively be created at any point in the application life cycle).

Since we already have a _create_pod method, can we have pvc() be a getter of a _pvc, which throws an error if not found (which is what we want...the program is being misused), and the setter be a private method (_set_pvc()) that is called in _create_pod?

Reply:
In general, I find one-way state change graphs much easier to reason about.

Reply:
I moved the lazy instantiation into the `_create_pod` method. I didn't add setters and getters because `_pvc` is private to the `Job` class.

Reply:
Got it thanks